### PR TITLE
Fir nose: Button component

### PR DESCRIPTION
**Pull request by modernserf**
http://fir-nose.glitch.me/#StoryUnstyledButton

A few remaining questions here:
- Does it make any sense for buttons to have an "inline" size that matches their context?
- Does the editor use any of the variants beyond "primary" and "secondary"? Do the "shadow" and "warning" buttons have a reason to exist?

@sheridanvk @ehmorris

### CODE REVIEW

File: lib/icon.js [hunk 1]
```diff
@@ -12,6 +13,7 @@ const IconBase = styled.img`
   display: inline-block;
   height: 1em;
   width: auto;
+  vertical-align: top;
```
Comment (modernserf) [inline lib/icon.js]: this is required for the buttons with icons to line up correctly with those that are without icons.

File: lib/icon.js [hunk 2]
```diff
@@ -330,7 +326,7 @@ export const StoryIcon_options = () => (
     <Inverted>
       {Object.keys(svgs).map((icon) => (
         <span key={icon}>
-          <Icon icon={icon} />
+          <Icon icon={icon} alt={icon} />
```
Comment (modernserf) [inline lib/icon.js]: Forgot to add this in the previous PR.

**Comment by ehmorris** [comment]
Aside from custom buttons, like the "Open in a New Window" button in the show menu, or the project pop button, we only have two button sizes, so I don't think an inline size makes sense.

We use primary, secondary, and tertiary, but the buttons are styled by context. For example, the buttons inside an `.info` div in a popover will be "tertiary" but the same button in the nav bar (like the sign-in button in the nav bar) will be "secondary". We use "primary" buttons in popovers, and we use "warning" buttons in popovers for delete actions, like "Delete This Project".

**Comment by modernserf** [comment]
What's the difference between a "secondary" and "tertiary" button?

**Comment by ehmorris** [comment]
@modernserf this was in storybook at some point:

Secondary:
Screen Shot 2019-07-29 at 1 15 52 PM

Tertiary:
Screen Shot 2019-07-29 at 1 15 58 PM

**Comment by modernserf** [comment]
so is "primary" just bigger? or do you mean that "primary" is the blue one?

**Comment by ehmorris** [comment]
@modernserf these three are primary:

Screen Shot 2019-07-29 at 1 49 10 PM

So there are three button styles in use:

1. Larger with a black border and text: primary
2. Smaller with a black border and text: secondary
3. Smaller with a gray border and text: tertiary

The teal button with a drop-shadow is a special style that we use only for a sign-up CTA in the editor.

**Comment by modernserf** [comment]
updates:
- removed the "shadow" variant, but included documentation how a consumer can create their own
- inlined the CTA colors into the button variant styles, since they are used nowhere else

**Comment by keithk** [comment]
:rocket: PR was released in [`v0.19.0`](https://github.com/glitchdotcom/shared-components/releases/tag/v0.19.0) :rocket:

**Pull request merged**